src/plots/cartesian/graph_interact.js

```diff
@@ -857,73 +857,95 @@ function createDroplines(hoverData, opts) {
         yAnchoredBase = yEdge - outerBBox.top,
         xBase = c0.ya.anchor === 'free' ? xFreeBase : xAnchoredBase,
         yBase = c0.xa.anchor === 'free' ? yFreeBase : yAnchoredBase,
-        color = c0.color;
+        xColor = c0.xa.spikecolor ? c0.xa.spikecolor : c0.color,
+        yColor = c0.ya.spikecolor ? c0.ya.spikecolor : c0.color,
+        xContrastColor = tinycolor(xColor).getBrightness() > 128 ?
+                '#000' : Color.background,
+        yContrastColor = tinycolor(yColor).getBrightness() > 128 ?
+                '#000' : Color.background,
+        xThickness = c0.xa.spikethickness,
+        yThickness = c0.ya.spikethickness,
+        xDash = Lib.dash(c0.xa.spikedash, xThickness),
+        yDash = Lib.dash(c0.xa.spikedash, yThickness),
+        xMarker = c0.xa.spikemode.indexOf('marker') !== -1,
+        yMarker = c0.ya.spikemode.indexOf('marker') !== -1,
+        xSpikeLine = c0.xa.spikemode.indexOf('toaxis') !== -1 || c0.xa.spikemode.indexOf('across') !== -1,
+        ySpikeLine = c0.ya.spikemode.indexOf('toaxis') !== -1 || c0.ya.spikemode.indexOf('across') !== -1,
+        xEndSpike = c0.xa.spikemode.indexOf('across') !== -1 ? xBase + xLength : xPoint,
+        yEndSpike = c0.ya.spikemode.indexOf('across') !== -1 ? yBase - yLength : yPoint;
 
     // Remove old dropline items
     container.selectAll('line.dropline').remove();
     container.selectAll('circle.dropline').remove();
 
 
     if(c0.ya.showspikes) {
-        // Background horizontal Line (to y-axis)
-        container.append('line')
-            .attr('x1', xBase)
-            .attr('x2', xPoint)
-            .attr('y1', yPoint)
-            .attr('y2', yPoint)
-            .attr('stroke-width', 5)
-            .attr('stroke', '#fff')
-            .attr('class', 'dropline');
-
-        // Foreground horizontal line (to y-axis)
-        container.append('line')
-            .attr('x1', xBase)
-            .attr('x2', xPoint)
-            .attr('y1', yPoint)
-            .attr('y2', yPoint)
-            .attr('stroke-width', 3)
-            .attr('stroke', color)
-            .attr('stroke-dasharray', '5,5')
-            .attr('class', 'dropline');
-
+        if(ySpikeLine) {
+            // Background horizontal Line (to y-axis)
+            container.append('line')
+                .attr('x1', xBase)
+                .attr('x2', xEndSpike)
+                .attr('y1', yPoint)
+                .attr('y2', yPoint)
+                .attr('stroke-width', yThickness + 2)
+                .attr('stroke', yContrastColor)
+                .attr('class', 'dropline');
```

Comment:
`.attr` can take an object ie `.attr({x1: xBase, x2: xEndSpike ... })`

also it's a bit nicer to use `.classed('dropline', true)` instead of `.attr('class', 'dropline')`, just because it makes things easier to extend later, if it turns out we need multiple classes or something.